### ISSUE TITLE
ERROR: Functionally annotate bins

### ISSUE THREAD

**Issue by xiaochen2015** [bxlab/metaWRAP#188]
Hi, I used annotate_bins module to annotate bins, but it went wrong. 
I reinstall the metawrap, but it didn't work. my metawrap version is 1.2.1
Details as follows: 
metawrap annotate_bins -o FUNCT_ANNOT -t 96 -b BIN_REASSEMBLY/reassembled_bins/

########################################################################################################################
########################################################################################################################

------------------------------------------------------------------------------------------------------------------------
-----                              setting up output folder and copything over bins...                             -----
------------------------------------------------------------------------------------------------------------------------

Warning: FUNCT_ANNOT already exists.
metawrap path: /home/xiaochen/miniconda2/envs/metawrap-env/bin/metawrap
Will use perl5 libraries located in /home/xiaochen/miniconda2/envs/metawrap-env/lib/perl5/site_perl/5.22.0 - hopefully they are there...
/home/xiaochen/miniconda2/envs/metawrap-env/bin/metawrap-scripts/shorten_contig_names.py BIN_REASSEMBLY/reassembled_bins//bin.10.strict.fa > FUNCT_ANNOT/tmp_bin.fa

------------------------------------------------------------------------------------------------------------------------
-----                                         NOW ANNOTATING bin.10.strict                                         -----
------------------------------------------------------------------------------------------------------------------------

prokka --quiet --cpus 96 --outdir FUNCT_ANNOT/prokka_out/bin.10.strict --prefix bin.10.strict FUNCT_ANNOT/tmp_bin.fa
blastp: error while loading shared libraries: libssl.so.1.0.0: cannot open shared object file: No such file or directory
[19:05:48] Could not determine version of blastp - please install version 2.2 or higher

************************************************************************************************************************
*****                        Something went wrong with annotating bin.10.strict. Exiting...                        *****
******************************************************************************************************

**Comment by ursky** [comment]
Are you using an isolated conda environment ("better installation")? 
If not, try installing it in a conda environment.
If yes, try running `ldd $(which openssl)` in the env to see where the sim link is? Then try running `conda install openssl=1.0` for force install `libssl.so.1.0.0`. Hopefully that works.

**Comment by xiaochen2015** [comment]
Thank you for your answer. it worked. 
According to "better installation", My metawrap was installed. "openssl" located at  ~/miniconda2/envs/metawrap-env/bin/openssl. And then running "conda install openssl=1.0", some softwares were updated, and some softwares (such as metawrap-mg 1.2.1 >1.2.0)were downgraded.

But  the CONCOCT endless warning messages in metaWRAP=1.2. It still has a bug.

**Comment by ursky** [comment]
Try running `conda install metawrap-mg=1.2.1 openssl=1.0` to enforce keeping both programs. This should resolve both problems. Let me know if this works for you. If it works, I'll enforce `openssl=1.0` in `metawrap=1.2.2`.

**Comment by xiaochen2015** [comment]
Yes, it worked. Thank you very much. It would be better to create a new environment for metawrap, and do not run it in the old environment.

**Comment by handibles** [comment]
Hey all, 

Had a similar(ish) issue with `openssl` and `opencrypto`:

```
> [...]
> [11:25:30] Could not determine version of blastp - please install version 2.2 or higher 
# hah? 
blastp
> blastp: error while loading shared libraries: libssl.so.1.0.0: cannot open shared object file: 
```

I had some previous compatibility issues where I had to spec `blast/openssl` versions, so was reluctant to change things in case they broke - in addition, I got a `not found / missing channel` when I gave it a look with `mamba install`. 

Instead - amended it with a symlink:`

`
ln -s /data/JPL/event_UAPs/nhi_host/sequencesandreversebio/bin/miniforge3/envs/mw.i/lib/libssl.so.1.1  /data/JPL/event_UAPs/nhi_host/sequencesandreversebio/bin/miniforge3/envs/mw.i/lib/libssl.so.1.0.0

ln -s /data/JPL/event_UAPs/nhi_host/sequencesandreversebio/bin/miniforge3/envs/mw.i/lib/libcrypto.so.1.1  /data/JPL/event_UAPs/nhi_host/sequencesandreversebio/bin/miniforge3/envs/mw.i/lib/libcrypto.so.1.0.0
`

Would be interested to hear whether @ursky & co. reckon symlinking like this is alright or not?...
